? usbasp.c
? usbasp.h
Index: Makefile.am
===================================================================
RCS file: /cvsroot/avrdude/avrdude/Makefile.am,v
retrieving revision 1.23
diff -u -r1.23 Makefile.am
--- Makefile.am	10 Feb 2005 14:54:57 -0000	1.23
+++ Makefile.am	9 Apr 2005 09:04:02 -0000
@@ -85,6 +85,8 @@
 	stk500.c \
 	stk500.h \
 	stk500_private.h \
+	usbasp.c \
+	usbasp.h \
 	term.c \
 	term.h
 
Index: avrdude.conf.in
===================================================================
RCS file: /cvsroot/avrdude/avrdude/avrdude.conf.in,v
retrieving revision 1.36
diff -u -r1.36 avrdude.conf.in
--- avrdude.conf.in	22 Dec 2004 01:30:30 -0000	1.36
+++ avrdude.conf.in	9 Apr 2005 09:04:10 -0000
@@ -243,6 +243,12 @@
   miso  = 10;
 ;
 
+programmer
+  id    = "usbasp";
+  desc  = "usbasp";
+  type  = usbasp;
+;
+
 # The programming dongle used by the popular Ponyprog
 # utility.  It is almost similar to the STK200 one,
 # except that there is a LED indicating that the
@@ -337,7 +343,6 @@
   miso  = 10;
 ;
 
-
 # From the contributor of the "xil" jtag cable:
 # The "vcc" definition isn't really vcc (the cable gets its power from
 # the programming circuit) but is necessary to switch one of the
Index: config_gram.y
===================================================================
RCS file: /cvsroot/avrdude/avrdude/config_gram.y,v
retrieving revision 1.39
diff -u -r1.39 config_gram.y
--- config_gram.y	22 Dec 2004 01:52:45 -0000	1.39
+++ config_gram.y	9 Apr 2005 09:04:11 -0000
@@ -34,6 +34,7 @@
 #include "pgm.h"
 #include "stk500.h"
 #include "avr910.h"
+#include "usbasp.h"
 #include "butterfly.h"
 #include "avr.h"
 
@@ -112,6 +113,7 @@
 %token K_SIZE
 %token K_STK500
 %token K_AVR910
+%token K_USBASP
 %token K_BUTTERFLY
 %token K_TYPE
 %token K_VCC
@@ -302,6 +304,11 @@
     }
   } |
 
+  K_TYPE TKN_EQUAL K_USBASP {
+    { 
+      usbasp_initpgm(current_prog);
+    }
+  } |
   K_TYPE TKN_EQUAL K_BUTTERFLY {
     { 
       butterfly_initpgm(current_prog);
Index: configure.ac
===================================================================
RCS file: /cvsroot/avrdude/avrdude/configure.ac,v
retrieving revision 1.22
diff -u -r1.22 configure.ac
--- configure.ac	10 Feb 2005 15:39:42 -0000	1.22
+++ configure.ac	9 Apr 2005 09:04:11 -0000
@@ -104,6 +104,21 @@
 AC_SUBST(SUBDIRS_AC, $SUBDIRS_AC)	
 AC_SUBST(DIST_SUBDIRS_AC, $DIST_SUBDIRS_AC)
 
+# USB stuff
+AH_TEMPLATE([HAVE_LIBUSB],
+	[Define if USB support is enabled via libusb])
+AC_CHECK_LIB(usb, main, have_libusb=yes)
+if test x$have_libusb = xyes; then
+   case $target in
+	*-*-darwin*)
+		LIBS="$LIBS -lusb -framework CoreFoundation -framework IOKit"
+		;;
+	*)
+		LIBS="$LIBS -lusb"
+		;;
+   esac
+   AC_DEFINE([HAVE_LIBUSB])
+fi
 
 # Find the parallel serial device files based on target system
 # If a system doesn't have a PC style parallel, mark it as unknown.
Index: lexer.l
===================================================================
RCS file: /cvsroot/avrdude/avrdude/lexer.l,v
retrieving revision 1.30
diff -u -r1.30 lexer.l
--- lexer.l	22 Dec 2004 01:52:45 -0000	1.30
+++ lexer.l	9 Apr 2005 09:04:12 -0000
@@ -156,6 +156,7 @@
 size             { yylval=NULL; return K_SIZE; }
 stk500           { yylval=NULL; return K_STK500; }
 avr910           { yylval=NULL; return K_AVR910; }
+usbasp           { yylval=NULL; return K_USBASP; }
 butterfly        { yylval=NULL; return K_BUTTERFLY; }
 type             { yylval=NULL; return K_TYPE; }
 vcc              { yylval=NULL; return K_VCC; }
